### PR TITLE
Validity

### PR DESCRIPTION
**Pull request by EvanBarbour3**
- Validation period has been added to installation. Can be changed on edit page.
- Validity period of application is dependent on the validity period in installations
- Requester and location added to application

### CODE REVIEW

File: app/Basket/Synchronisation/ApplicationSynchronisationService.php [hunk 1]
```diff
@@ -204,7 +205,8 @@ public function initialiseApplication(
         $validity,
         $productGroup,
         array $productOptions,
-        $location
+        $location,
+        $requester
```
Comment (wnowicki) [inline app/Basket/Synchronisation/ApplicationSynchronisationService.php]: missing DocBlock

File: app/Basket/Installation.php [hunk 2]
```diff
@@ -52,6 +53,7 @@ class Installation extends Model
         'ext_notification_url',
         'ext_default_product',
         'location_instruction',
+        'validity',
```
Comment (wnowicki) [inline app/Basket/Installation.php]: what about validation in controller?

File: app/Basket/Synchronisation/ApplicationSynchronisationService.php (hunk 1 above)
Comment (EvanBarbour3) [inline app/Basket/Synchronisation/ApplicationSynchronisationService.php]: Added

File: app/Basket/Installation.php (hunk 2 above)
Comment (EvanBarbour3) [inline app/Basket/Installation.php]: Validation added to installations controller

**Pull request merged**